django/db/migrations/serializer.py

```diff
@@ -246,11 +247,14 @@ class RegexSerializer(BaseSerializer):
     def serialize(self):
         imports = {"import re"}
         regex_pattern, pattern_imports = serializer_factory(self.value.pattern).serialize()
-        regex_flags, flag_imports = serializer_factory(self.value.flags).serialize()
+        # We turn off default implicit flags, because regexes with the same
+        # implicit and explicit flags are not equal.
+        flags = self.value.flags ^ re.compile('').flags
```

Comment:
Would it make sense to set implicit flags along with implicit flags or is that exactly what's not working?